SUS-1092 | Forum refactor by sqreek

macbre opened the pull request:
See #11566 / https://wikia-inc.atlassian.net/browse/SUS-1092

This reverts commit 3ed0ec9265df4591317632cf83b93c6a17866cc4, reversing
changes made to 898fb96fedc9bd7918b433eb65cfc041e1365e65.

Conflicts:
	extensions/wikia/Forum/ForumController.class.php
	extensions/wikia/Wall/WallBaseController.class.php

macbre committed:
Revert "Merge pull request #11653 from Wikia/revert-11600-revert_revert_merge_wall_refactor_with_fix"

See #11566

This reverts commit 3ed0ec9265df4591317632cf83b93c6a17866cc4, reversing
changes made to 898fb96fedc9bd7918b433eb65cfc041e1365e65.

Conflicts:
	extensions/wikia/Forum/ForumController.class.php
	extensions/wikia/Wall/WallBaseController.class.php

rybmat commented on extensions/wikia/Forum/ForumBoardInfo.class.php:
```diff
@@ -0,0 +1,63 @@
+	private $lastPost;
+
+	/**
+	 * ForumBoardInfo constructor.
```
does this comment give us any value?

mixth-sense replied:
No. Feel free to remove it. It is ridiculous. :-)

rybmat commented on extensions/wikia/Wall/WallBaseController.class.php:
```diff
@@ -635,21 +621,31 @@ protected function checkAndSetUserBlockedStatus( $wallOwner = null ) {
+		}
 
+		wfRunHooks( 'WallGreetingContent', [ &$greetingText ] );
+		return $greetingText; // used by SWM to add messages to Wall in monobook
```
SWM or SMW?

macbre replied:
It's Site Wide Messages, I suppose :) @sqreek ?

mixth-sense replied:
It is Site Wide Messages, correct.

mixth-sense replied:
Although, it is unclear how exactly Site Wide Messages use this thing.

rybmat commented on extensions/wikia/Forum/ForumBoard.class.php:
```diff
@@ -120,12 +77,41 @@ public function getTotalActiveThreads( $relatedPageId = 0, $db = DB_SLAVE ) {
 	 */
 	public function clearCacheBoardInfo() {
-		$this->getBoardInfo( DB_MASTER );
+		$this->getBoardInfoFromMaster();
```
personally I don't like situation when the method name is getSomething and instead of using the returned value we are calling it only because of side effects (cache in this example) But TBH I have no better proposition for name for this method

mixth-sense replied:
I would use `load` instead of `get`. I agree that `get` implies that the method will  go and `get` you something as the returned value. As for better name `load` seems good in this context. We are in the scope of `ForumBoard` and all we need is to `load` the `BoardInfo`.

rybmat commented on extensions/wikia/Forum/ForumController.class.php:
```diff
@@ -31,30 +31,20 @@ public function board() {
-			$this->response->addAsset( 'extensions/wikia/Forum/css/monobook/ForumBoardMonobook.scss' );
-		}
-
-		$this->addMiniEditorAssets();
```
I wonder if this really needs to be separate method

rybmat commented on extensions/wikia/Forum/ForumController.class.php:
```diff
@@ -113,21 +103,31 @@ public function getWallForIndexPage( $title ) {
 		}
 
-		// MAIN-7865: Don't render MiniEditor on Monobook
-		$this->response->setVal( 'showMiniEditor', $this->wg->EnableMiniEditorExt && $this->app->checkSkin( 'oasis' ) );
```
are we sure that this should be removed?

mixth-sense replied:
I think it is an error. MiniEditor is actively used and lack of the value would cause any checks evaluate to `false` unless a default is set.

UPDATE: I don't see any default to be set anywhere. Only `if ($showMiniEditor)` checks.

rybmat commented on extensions/wikia/Forum/ForumController.class.php:
```diff
@@ -356,4 +359,23 @@ public function forumRelatedThreads() {
+			$this->response->addAsset( 'extensions/wikia/Forum/css/monobook/ForumBoardMonobook.scss' );
+		}
+
+		$this->addMiniEditorAssets();
```
I wonder if this really needs to be separate method

mixth-sense replied:
It does not. It is used only once and all it does is a side effect. It is not something we can test in a sane way and I do not think a separare `addMiniEditorAssets()` makes `addAssets()` less or more testable. It is simply a piece of code written for IDE rather than for a human. Pointless. :(

macbre commented on extensions/wikia/Forum/ForumBoard.class.php:
```diff
@@ -120,12 +77,41 @@ public function getTotalActiveThreads( $relatedPageId = 0, $db = DB_SLAVE ) {
 	public function clearCacheBoardInfo() {
-		$this->getBoardInfo( DB_MASTER );
+		$this->getBoardInfoFromMaster();
 		$this->getTotalActiveThreads( DB_MASTER );
```
I'd avoid calling these two methods and simply delete memcache keys that are used by these two methods above.

Let's untangle this :spaghetti: :smile:

macbre commented on extensions/wikia/Forum/Forum.class.php:
```diff
@@ -286,7 +280,7 @@ protected function createOrEditBoard( $board, $titletext, $body, $bot = false )
 
 		Forum::$allowToEditBoard = false;
-
+		$board->clearCacheBoardInfo();
```
```
Unexpected non-MediaWiki exception encountered, of type "Error"
Error: Call to a member function clearCacheBoardInfo() on null in /usr/wikia/slot1/16124/src/extensions/wikia/Forum/Forum.class.php:283
```

A check similar to the  one in line 242 may be useful here.

macbre committed:
Forum: fix a fatal error

Error: Call to a member function clearCacheBoardInfo() on null in /usr/wikia/slot1/16124/src/extensions/wikia/Forum/Forum.class.php:283

macbre commented on extensions/wikia/Forum/ForumBoard.class.php:
```diff
@@ -134,4 +120,26 @@ public function clearCacheBoardInfo() {
 		}
 	}
 
+	private function getLastPost( $rev_id ) {
```
Let's document what kind of values this method returns `ForumPostInfo|null`

macbre commented on extensions/wikia/Forum/ForumBoardInfo.class.php:
```diff
@@ -0,0 +1,63 @@
+		$this->threadCount = $threadCount;
+	}
+
+	public function setLastPost( $lastPost ) {
```
Let's force the class of the argument - `ForumPostInfo`.

macbre commented on extensions/wikia/Forum/ForumController.class.php:
```diff
@@ -356,4 +359,23 @@ public function forumRelatedThreads() {
+
+	public function messageTopic() {
+		// stub function
+	}
```
I do not see any usage of this method.

macbre replied:
`Forum_boardNewThread.php` makes the following class:

```php
<?= F::app()->renderPartialCached( 'Wall', 'messageTopic', [ ] ) ?>
```

macbre committed:
ForumController: remove unused messageTopic() method

macbre committed:
Add type hints

macbre committed:
ForumBoard: delete memcache entries only when calling clearCacheBoardInfo()

rybmat committed:
SUS-1092: revert removal of setting showMiniEditor

macbre committed:
ForumBoardInfo: address CR comments

macbre committed:
WallBaseController: get rid of unused WallGreetingContent hook

macbre committed:
ForumController: no need to have a separate addMiniEditorAssets method

macbre committed:
ForumBoard: prevent a fatal

TypeError: Argument 1 passed to ForumBoardInfo::setLastPost() must be an instance of ForumPostInfo, null given, called in /usr/wikia/slot1/16143/src/extensions/wikia/Forum/ForumBoard.class.php on line 99 and defined in /usr/wikia/slot1/16143/src/extensions/wikia/Forum/ForumBoardInfo.class.php:30

macbre committed:
ForumBoardInfo: kill a notice

PHP Notice: Undefined index: lastPost in /extensions/wikia/Forum/ForumBoardInfo.class.php on line 15
